<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrate methods: PANDA, LIONESS, CONDOR, ALPACA, SAMBAR, MONSTER and OTTER into one workflow • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Integrate methods: PANDA, LIONESS, CONDOR, ALPACA, SAMBAR, MONSTER and OTTER into one workflow">
<meta property="og:description" content="PANDA(Passing Attributes between Networks for Data Assimilation) is a message-passing model to reconstruction gene regulatory network. It integrates multiple sources of biological data, including protein-protein interaction data, gene expression data, and sequence motif information to reconstruct genome-wide, condition-specific regulatory networks.[(Glass et al. 2013)]. LIONESS(Linear Interpolation to Obtain Network Estimates for Single Samples) is a method to estimate sample-specific regulatory networks by applying linear interpolation to the predictions made by existing aggregate network inference approaches. CONDOR(COmplex Network Description Of Regulators)is a bipartite community structure analysis tool of biological networks, especially eQTL networks, including a method for scoring nodes based on their modularity contribution.[(Platig et al. 2016). ALPACA(ALtered Partitions Across Community Architectures) is a method for comparing two genome-scale networks derived from different phenotypic states to identify condition-specific modules.[(Padi and Quackenbush 2018)]. This package integrates pypanda--the Python implementation of PANDA and LIONESS(https://github.com/davidvi/pypanda),the R implementation of CONDOR(https://github.com/jplatig/condor) and the R implementation of ALPACA (https://github.com/meghapadi/ALPACA) into one workflow. Each tool can be call in this package by one function, and the relevant output could be accessible in current R session for downstream analysis. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/ALPACA.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="articles/ApplicationinGTExData.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="articles/ApplicationwithTBdataset.html">netZooR application with TB dataset</a>
    </li>
    <li>
      <a href="articles/CONDOR.html">Using CONDOR for community detection in bipartite graphs</a>
    </li>
    <li>
      <a href="articles/EGRET_toy_example.html">Constructing genotype-specific gene regulatory networks with EGRET</a>
    </li>
    <li>
      <a href="articles/LionessApplicationinGTExData.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="articles/MONSTER.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="articles/SAMBAR.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="articles/TutorialOTTER.html">Inferring Gene Regulatory Networks from GTEx Gene Expression Data in R with OTTER</a>
    </li>
    <li>
      <a href="articles/pandaR.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="articles/pandaRApplicationinGTExData.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">


<p><a href="mailto:R4.01@Ubuntu" class="email">R4.01@Ubuntu</a> 18.04: <a href="http://zookeeper.networkmedicine.org/job/netZooR_Ubuntu18.04/"><img src="http://zookeeper.networkmedicine.org/buildStatus/icon?job=netZooR_Ubuntu18.04" alt="Build Status"></a></p>
<div id="description" class="section level2">
<h2 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h2>
<p>netZooR is an R package to reconstruct, analyse, and plot biological networks.</p>
</div>
<div id="features" class="section level2">
<h2 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h2>
<p>netZooR currently integrates:</p>
<ul>
<li><p><strong>PANDA</strong> (Passing Attributes between Networks for Data Assimilation) <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0064832">[Glass et al. 2013]</a>: constructs gene regulatory network from gene expression data, protein-protein interaction data, and transcription factor binding motifs (TFBMs) data.</p></li>
<li><p><strong>CONDOR</strong> (COmplex Network Description Of Regulators) <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005033">[Platig et al. 2016]</a>: analyses bipartite community structure analysis of biological networks.</p></li>
<li><p><strong>LIONESS</strong> (Linear Interpolation to Obtain Network Estimates for Single Samples) <a href="https://doi.org/10.1016/j.isci.2019.03.021">[Kuijjer et al. 2019]</a>: reconstructs single-sample gene regulatory networks.</p></li>
<li><p><strong>ALPACA</strong> (ALtered Partitions Across Community Architectures) <a href="https://www.nature.com/articles/s41540-018-0052-5">[Padi and Quackenbush 2018]</a>: compares two networks and identify changes in modular structure.</p></li>
<li><p><strong>SAMBAR</strong> (Subtyping Agglomerated Mutations By Annotation Relations) <a href="https://www.nature.com/articles/s41416-018-0109-7">[Kuijjer et al.]</a>: identifies subtypes based on somatic mutation data.</p></li>
<li><p><strong>MONSTER</strong> (Modeling Network State Transitions from Expression and Regulatory data) <a href="https://doi.org/10.1186/s12918-017-0517-y">[Schlauch et al.]</a>: infers transcription factor drivers of cell state conditions at the gene regulatory network level.</p></li>
<li><p><strong>OTTER</strong> (Optimization to Estimate Regulation) <a href="https://www.biorxiv.org/content/10.1101/2020.06.23.167999v2.abstract">[Weighill et al.]</a>: models gene regulation estimation as a graph matching problem.</p></li>
<li><p><strong>CRANE</strong> (Constrained Random Alteration of Network Edges) <a href="https://www.biorxiv.org/content/10.1101/2020.07.12.198747v1">[Lim et al.]</a>: generates ensembles of gene regulatory networks to identify disease modules.</p></li>
<li><p><strong>EGRET</strong> (Estimating the Genetic Regulatory effects on TFs) <a href="">[Weighill et al.]</a> In preparation: models individual-specific gene regulatory networks using their genetic variants.</p></li>
</ul>
<p>The package also integrates additional functions to:</p>
<ul>
<li><p>Source protein-protein interaction network from <a href="https://string-db.org/">STRINGdb</a> based on a list of protein of interest.</p></li>
<li><p>Plot one PANDA network in <a href="https://cytoscape.org/">Cytoscape</a>.</p></li>
<li><p>Plot two differential PANDA networks in Cytoscape.</p></li>
</ul>
</div>
<div id="requirements-installation-and-basic-configuration" class="section level2">
<h2 class="hasAnchor">
<a href="#requirements-installation-and-basic-configuration" class="anchor"></a>Requirements, installation and basic configuration.</h2>
<ul>
<li><p>netZooR is compatible with R (&gt;= 3.3.3) including R (&gt;= 4.0), click <a href="https://www.r-project.org/">here</a> for more installation details.</p></li>
<li>
<p>To use PANDA and LIONESS, there are two options:</p>
<ol>
<li>
<p>Use <code><a href="reference/panda.py.html">panda.py()</a></code> and <code><a href="reference/lioness.py.html">lioness.py()</a></code> by invoking the respective Python implementations in <a href="(https://github.com/netZoo/netZooPy/tree/netZoo)">netZooPy</a>. Because the native R linear algebra libraries can be slow, this way is recommended for faster analysis. However, optimized parallel libraries can give reasonable run times (option ii). To invoke Python scripts, there are some requirements to meet before using netZooR:</p>
<p>a) <a href="https://www.python.org/downloads/"><strong>Python</strong></a> (&gt;= 3.5.0) installed;</p>
<p>b) Python libraries <a href="https://pandas.pydata.org/">pandas</a>, <a href="https://numpy.org/">numpy</a>, and <a href="https://www.scipy.org/">scipy</a> installed;</p>
<p>c) Internet access as package <code>reticulate</code> will link the R wrapper to the Python scripts located <a href="https://github.com/netZoo/netZooPy/tree/netZoo">here</a> for those two methods.</p>
</li>
<li>
<p>Use <code>panda()</code> and <code><a href="reference/lioness.html">lioness()</a></code> for the pure R implementations of PANDA and LIONESS. To speed up the run time, it is highly recommended to install an optimized linear algebra library, particularly for Ubunutu. Macos generally comes with optimized linear algebra libraries. You can check the <code>BLAS/LAPACK</code> fields in <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> in your R console. Detailed instructions can be found <a href="https://csantill.github.io/RPerformanceWBLAS/">here</a>.</p>
<p><span class="emoji" data-emoji="warning">⚠️</span> However, we found that Intel MKL linear algebra library with R 4.0.3 on Ubuntu 18.04 gave inconsistent results for the multiplication of large matrices and the results of PANDA were inconsistent. Therefore, Intel MKL is not currently recommended.</p>
</li>
</ol>
</li>
<li><p>Most of plotting function can be realized by functions in <a href="https://igraph.org/redirect.html">igraph</a>, which will be loaded with netZooR through <code>library(netZooR)</code>. Some plotting functions like <code><a href="reference/vis.panda.in.cytoscape.html">vis.panda.in.cytoscape()</a></code> and <code><a href="reference/vis.diff.panda.in.cytoscape.html">vis.diff.panda.in.cytoscape()</a></code> are able to plot interactive PANDA networks in <a href="https://cytoscape.org/">Cytoscape</a>, but installation of Cytoscape is required before using these plotting functions. Also, please make sure that Cytoscape is open when these functions are called.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co"># install.packages("devtools") </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co"># install netZooR pkg with vignettes, otherwise remove the "build_vignettes = TRUE" argument.</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"netZoo/netZooR"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">netZooR</span><span class="op">)</span></pre></div>
<p>For more details please refer to the <a href="https://netzoo.github.io/netZooR/">documentation website</a>.</p>
<p>This package will invoke Python programming language in R environment through <a href="https://rstudio.github.io/reticulate/">reticulate</a> package, by default setting there is no additional configuration needed. Configuring which version of Python to use , here in netZooR, Python 3.X is required. More details can be found <a href="https://cran.r-project.org/web/packages/reticulate/vignettes/versions.html">here</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">#check your Python configuration and the specific version of Python in use currently</span>
<span class="fu">py_config</span><span class="op">(</span><span class="op">)</span>

<span class="co"># reset to Python 3.X if necessary, like below:</span>
<span class="co"># use_python("/usr/local/bin/python3")</span></pre></div>
</div>
<div id="issues" class="section level2">
<h2 class="hasAnchor">
<a href="#issues" class="anchor"></a>Issues</h2>
<p>For <strong>data.table</strong> installation issue please refer to <a href="https://github.com/netZoo/netZooR/issues/40">issue #40</a>.</p>
<p>Please report any further issue to the <a href="https://github.com/netZoo/netZooR/issues">issues page</a>.</p>
</div>
<div id="tutorials" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorials" class="anchor"></a>Tutorials</h2>
<p>For more details please refer to the <a href="https://netzoo.github.io/netZooR/">documentation website</a>. Tutorials are available at the top navigation bar <strong>Articles/</strong> for basic usage and application cases. Or use <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("netZooR")</a></code> after installing the package. Also check <a href="http://netbooks.networkmedicine.org">netbooks</a> to go through the tutorials on a jupyter notebook cloud server.</p>
</div>
<div id="contribution-and-development" class="section level2">
<h2 class="hasAnchor">
<a href="#contribution-and-development" class="anchor"></a>Contribution and Development</h2>
<p>Contributions are welcome! The contribution guide to netZooR can be found <a href="https://netzoo.github.io/contribute/contribute/">here</a>. After adding new features or optimizing a function in the package, please re-build the package and run <code>R CMD check .</code> in the terminal or follow the instructions below before doing the pull request to the devel branch.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://pkgdown.r-lib.org">pkgdown</a></span><span class="op">)</span>
<span class="co"># document the description of function</span>
<span class="co"># setwd('path/to/netZooR/root') # Set the working directory to netZooR root</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/document.html">document</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># build vignettes</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/build_vignettes.html">build_vignettes</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># You can skip building the vignettes if you are not contributing a vignette</span>
<span class="co"># build documentation website</span>
<span class="fu">pkgdown</span><span class="fu">::</span><span class="fu"><a href="https://pkgdown.r-lib.org/reference/build_site.html">build_site</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Install and build the package using devtools</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/install.html">install</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># To install the dependecies</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/build.html">build</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># To build the package</span>
<span class="co">#devtools::build(vignettes = FALSE) # You can skip building the vignettes if you are not contributing a vignette</span>

<span class="co"># CMD check, if passed all tests here, it means this package is ready to pull request to the devel branch. Otherwise, fix the bug before pulling request.</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/check.html">check</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#devtools::check(vignettes = FALSE) #You can skip building the vignettes if you are not contributing a vignette</span></pre></div>
<p>The master branch on github should always be in good shape, so please to pull request to the <strong>devel</strong> branch. If the contribution is specific to pandaR, please contribute to its seperate GitHub page by <a href="https://github.com/jnpaulson/pandaR">pull request</a>.</p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h2>
<p>The software is free and is licensed under the GNU General License v3.0, see the file <a href="LICENSE">LICENSE</a> for details.</p>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of conduct</h2>
<p>Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Tian Wang <br><small class="roles"> Author, maintainer </small>  </li>
<li>Marouen Ben Guebila <br><small class="roles"> Author, contributor </small>  </li>
<li>John Platig <br><small class="roles"> Author </small>  </li>
<li>Marieke Kuijjer <br><small class="roles"> Author </small>  </li>
<li>Magha Padi <br><small class="roles"> Author </small>  </li>
<li>Rebekka Burkholz <br><small class="roles"> Author </small>  </li>
<li>Deborah Weighill <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="http://zookeeper.networkmedicine.org/job/netZooR_Ubuntu18.04/"><img src="http://zookeeper.networkmedicine.org/buildStatus/icon?job=netZooR_Ubuntu18.04" alt="Build Status"></a></li>
<li><a href="https://codecov.io/gh/netZoo/netZooR"><img src="https://codecov.io/gh/netZoo/netZooR/branch/devel/graph/badge.svg" alt="codecov"></a></li>
<li><a href="https://www.gnu.org/licenses/gpl-3.0"><img src="https://img.shields.io/badge/License-GPLv3-blue.svg" alt="License: GPL v3"></a></li>
<li><a href="https://netzoo.github.io/netZooR/"><img src="https://img.shields.io/badge/docs-passing-green"></a></li>
<li><a href="https://github.com/netZoo/netZooR/tree/master/vignettes"><img src="https://img.shields.io/badge/netZooR-tutorials-9cf" alt="tutorials"></a></li>
<li><a href="http://netbooks.networkmedicine.org/user/marouenbg/notebooks/Welcome_to_netBooks.ipynb?"><img src="https://img.shields.io/badge/netZooR-netBooks-ff69b4" alt="netBooks"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz, Deborah Weighill.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
